test/models/fraction.spec.ts

```diff
@@ -137,4 +137,15 @@ describe("Fraction", () => {
       assert.equal(f1.sub(f2).toNumber(), 0.00000001);
     });
   });
+
+  describe("toNumber", () => {
+    it("Can serialize large coprime Fractions", () => {
+      // Using two prime number ~10**18
+      const f = new Fraction(
+        new BN("1000000000000000003", 10),
+        new BN("1000000000000000009", 10)
```

Comment:
Would this still throw if the denominator were `1` for example? I would just expect there to be a loss of precision in that case.

Reply:
It will actually become infinity as we shift away the 1 to 0. I guess that's fine? Will add a unit test to document.

Reply:
I'd expect it to be equal to `1e18` as a floating point number. I would expect it to just lose precision but not be `Infinity`. For example in JS:
```
> 1000000000000000009
1000000000000000000
```
Because of precision loss.

Reply:
Do you have an idea of how to achieve this. I cannot call `.toNumber()` on a BN that has more than 53 bits. I would have to call `toString()` and then `parseInt()` or something hacky like that.

Reply:
Something like this: https://github.com/gnosis/ethcontract-rs/blob/master/src/conv.rs#L5-L14

This will give you two doubles that you can divide.